### ISSUE TITLE
Vario Gauge - Graphical Issues

### ISSUE THREAD

**Issue by lordfolken** [XCSoar/XCSoar#1396]
Original
![image](https://github.com/XCSoar/XCSoar/assets/407371/6adcacd4-c885-4346-ba5a-8f6e8f221c46)

WIP:
![image](https://github.com/XCSoar/XCSoar/assets/407371/4a4d7d4a-45e8-46cd-8238-987affbf3766)

Issues:
 - [x] Labels NetAvg / Gross / MC are maligned e5b2ffc95403bb19d2ba264fcb5e10ce22b69dbf
 - [x] The arc labels overflow the box 964d45e532def1368d67ac31532479c433d47db6
 - [x] The values under NetAvg/Gross/MC do not resize, the labels themselves do. 21fe2d0caa3a9d5e087f3e66d96a6304b08fbbd8
 - [x] The values should be large and the labels small 21fe2d0caa3a9d5e087f3e66d96a6304b08fbbd8
 - [ ] arc is overlapping rc

I think a design along the butterfly vario, would be dooable and look good.

![image](https://github.com/XCSoar/XCSoar/assets/407371/d6721b23-69fc-414c-b5c9-f532057801cc)

**Comment by DanD222** [comment]
I remember seeing the vario scale replaced by vector lines a while ago on the forum, might be worth looking into as well? The vario scale looks a lot better this way IMO.

https://forum.xcsoar.org/viewtopic.php?p=17530#p17530

[0001-Replace-bitmap-vario-scale-by-drawn-lines.patch.txt](https://github.com/XCSoar/XCSoar/files/14896109/0001-Replace-bitmap-vario-scale-by-drawn-lines.patch.txt)

**Comment by lordfolken** [comment]
None of this is a  bitmap. I think that patch was merged long ago.

**Comment by DanD222** [comment]
Here’s the photo from the forum:
![IMG_6491](https://github.com/XCSoar/XCSoar/assets/48011765/ca7960f5-0f7d-4e9e-bf37-96aa2b3f122b)

It doesn’t look very similar to the current implementation to me.

> None of this is a bitmap

You mean the two vario scales in your first post?

**Comment by lordfolken** [comment]
Yes. Neither scala is a bitmap. Its just maths and then rendered. The bitmap scale was removed some time ago.
Ill have a look at the patch? Maybe there is some inspiration there

**Comment by DanD222** [comment]
OK, thanks. Maybe see how it looks without the arc? (that doesn’t look all that great IMO)

**Comment by lordfolken** [comment]
@DanD222 please review.

**Comment by DanD222** [comment]
Some first thoughts.. 

- Looking at the B&W WIP screenshot in your first post, I think that the horizontal line at 0 lift/sink could be made shorter, so as to be the same length as the other lift/sink marks.

- It seems to me that the lines at -3 and -4 are oriented towards a different point in the centre of the now imaginary arc than the lines at -1 and -2.

- I'm not sure if it's doable to have the grey scale lift/sink numbers be "forward" of the white arrow?

- I'm thinking whether the green arc shouldn't be more on the outside od the lift marks (possibly move the lift marks + labels more inwards so that the green bar doesn't run out of space)? Alternatively, are we able to have the green arc "behind" the white lift marks?

- I'm also thinking about is whether the arrows shouldn't be touching/referenced to the right edge of the lines/green bar, instead of the current left edge. This might necessitate moving the numbers inwards a bit.

- I would also consider making the NetAvg hollow arrow a bit thicker.


I don't use the vario display myself, so I'm not all that familiar with it, and had to think about what's displayed for a moment. I'm not sure if this is within the scope of what you're trying to improve, but:

- The MC arrow is visually almost the most prominent element, while being the least useful one (compare to the tiny "Little gray pointer (reverse) with tip" on the Butterfly). I initially thought the MC arrow was the instantaneous/gross vario indicator.

Is that something we can also adjust perhaps?  Could we colour-code the values in the centre of the vario scale the same way as the arrows on the vario (i.e. NetAvg, Gross, MC), so that there are visual cues as to which indicator is displaying which value, providing there's enough contrast so as to be able to differentiate between them on a B&W e-ink display?

**Comment by elgandoz** [comment]
Sorry to jump in, I would like to throw my 2 cents here... I quickly realised a mockup to test an improved display on the gauge, a bit more concentrical / symmetrical (like in the forum pic), using a more detailed scale (like the butterfly vario, or a classic analog one - see belove pic)
![vario](https://github.com/XCSoar/XCSoar/assets/5859926/ed126cf1-dcd8-4902-bd14-aa817874e5f0)

The scale would work for both m/s and kt and it could be a more readable.
Also, just like the showcased varios, I believe there is no need to add the minus ["-"] sign in the negative scale, it just add noise in a small area.

<img src="https://github.com/XCSoar/XCSoar/assets/5859926/bf3245be-b014-47f7-bfa2-9738d134ec4f" width="50%" alt="Gauge" />

Given also the tiny space and the dimension of the scale numbers on a real screen, we could also consider to remove the scale numbers since the values are already visible within the Gross infobox at the center
<img src="https://github.com/XCSoar/XCSoar/assets/5859926/f1567c21-2638-43a8-9c5c-b0b5a37060ac" width="50%" alt="Gauge no scale" />

I'm not sure how easy this would be in code, I'm just trying to suggest how to improve the readability.
